securitycenter/assets/main.go

```diff
@@ -0,0 +1,220 @@
+
+	listAssetsRequest := &securitycenterpb.ListAssetsRequest{}
+	// orgId is the numeric organization ID.  e.g. 01231231
+	listAssetsRequest.Parent = fmt.Sprintf("organizations/%s", orgId)
+	listAssetsRequest.Filter = "security_center_properties.resource_type=\"google.cloud.resourcemanager.Project\""
+	// Convert the time to a Timestamp protobuf
+	readTime, err := ptypes.TimestampProto(asOf)
+	if err != nil {
+		fmt.Printf("Error converting %v: %v", asOf, err)
+		return 0, err
+	}
+	listAssetsRequest.ReadTime = readTime
+
+	it := client.ListAssets(ctx, listAssetsRequest)
+	var assetsFound = 0
+	// Process the results from the iterator.
+loop:
+	for {
+		// r is a ListAssetsResult object.  See API reference for
+		// details on its fields.
+		r, err := it.Next()
+		switch err {
+		case nil:
+			fmt.Printf("Asset Name: %s, Resource Name %s, Resource Type %s\n", r.Asset.Name, r.Asset.SecurityCenterProperties.ResourceName, r.Asset.SecurityCenterProperties.ResourceType)
+
+			assetsFound++
+		case iterator.Done:
+			break loop
+		default:
+			fmt.Printf("Error listing assets: %v", err)
+			return 0, err
+		}
+	}
+	return assetsFound, nil
+}
+
+// [END list_project_assets_at_time]
+
+// List all assets in an organization that are GCP projects along with
+// there state change from 30 days ago.
+// [START list_project_assets_with_state_changes]
+func ListAllProjectAssetsWithStateChanges(orgId string) (int, error) {
+	// Initialize a new context and client and return any errors (e.g
+	// credentials were not found)
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		fmt.Printf("Error instantiating client %v\n", err)
+		return 0, err
+	}
+
+	listAssetsRequest := &securitycenterpb.ListAssetsRequest{}
+	// orgId is the numeric organization ID.  e.g. 01231231
+	listAssetsRequest.Parent = fmt.Sprintf("organizations/%s", orgId)
+	listAssetsRequest.Filter = "security_center_properties.resource_type=\"google.cloud.resourcemanager.Project\""
+	// Convert the time (10 days) to a Duration protobuf.
+	daysAgo, _ := time.ParseDuration("240h")
+	fmt.Printf("Days ago: %v\n", daysAgo)
+	// Compare all current assets to there state as of 10 days ago.
+	listAssetsRequest.CompareDuration = ptypes.DurationProto(daysAgo)
+
+	it := client.ListAssets(ctx, listAssetsRequest)
+	var assetsFound = 0
+	// Process the results from the iterator.
+loop:
+	for {
+		// r is a ListAssetsResult object.  See API reference for
+		// details on its fields.
+		r, err := it.Next()
+		switch err {
+		case nil:
+			fmt.Printf("Asset Name: %s, Resource Name %s, Resource Type %s, State Change %s\n", r.Asset.Name, r.Asset.SecurityCenterProperties.ResourceName, r.Asset.SecurityCenterProperties.ResourceType, r.State)
+
+			assetsFound++
+		case iterator.Done:
+			break loop
+		default:
+			fmt.Printf("Error listing assets: %v", err)
+			return 0, err
+		}
+	}
+	return assetsFound, nil
+}
+
+// [END list_project_assets_with_state_changes]
+
+func main() {
```

Comment:
Samples should not be a `package main` (with a `func main`) unless they're a quickstart. In the past, we've had issues keeping binaries up to date unless they're part of documentation. Tests should be used to ensure the samples work.